File: platform/Leetcode/src/split_balanced_string.py
def split_balanced_string(string):
    '''
    Function returns the maximum number of balanced groupings of L & R strings
    e.g., Input: s = "LLLLRRRR". Outputs 1, because 'LLLLRRRR' is one group
    '''
    assert_message = "Input must be a balanced combination of L & R"
    assert len(set(string)) == 2, assert_message
    assert string.count(list(set(string))[0]) == string.count(list(set(string))[-1]), assert_message
    
    total_cnt = 0
    leader_counter = 0
    
    ind = 0
    leader = string[0]
    while ind < len(string):
        if string[ind] == leader:
            leader_counter += 1            
        else:
            leader_counter -= 1            

        if leader_counter == 0:
            total_cnt += 1

        ind += 1
    
    return total_cnt

def main():
    text = input("Enter an interspersed string of L's and R's: ")
    output = split_balanced_string(text)
    print('The max number of balanced LR-group is: ', output)

if __name__ == '__main__':
    main()

File: platform/Leetcode/src/SplitBalancedString.java
import java.util.Arrays;
import java.util.ArrayList;
import java.util.Scanner;

public class SplitBalancedString {
    public static void main (String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.print("Enter an interspersed string of L's and R's : ");
        String input = scan.nextLine();
        
        int output = splitBalancedString(input);
        System.out.println("The maximum number of LR-grouping is :" + output);
        
    }
    
    public static int splitBalancedString(String something) {
        ArrayList<String> letterArray = new ArrayList<String>(Arrays.asList(something.split("")));
        int total_counter = 0;
        int leader_counter = 0;


        String leader = String.valueOf(something.charAt(0));
        int ind = 0;
        while (ind < letterArray.size()) {

            String let = letterArray.get(ind);
            if (let.equals(leader)) {
                leader_counter +=1;  
            } else {
                leader_counter -= 1;
            }

            if (leader_counter == 0) {
                total_counter += 1;
                }
            ind++;

        }
        return total_counter;
    }
    
}